core/Dispatch.cpp

```diff
@@ -31,38 +32,96 @@ namespace olympia
 
         auto execution_topology = coreutils::getExecutionTopology(node->getParent());
 
-        // Create Disptchers for ALU, FPU, BR -- one to many of them
-        // depending on the execution_topology extension
-        for (auto exe_unit_pair : execution_topology)
+        auto issue_queue_pipe_topology = olympia::coreutils::getPipeTopology(node->getParent(), "issue_queue_topology");
+
+        for (size_t i = 0; i < issue_queue_pipe_topology.size(); i++)
         {
-            const auto tgt_name   = exe_unit_pair[0];
-            const auto unit_count = exe_unit_pair[1];
+            auto exe_units = issue_queue_pipe_topology[i];
+            auto exe_unit_first = exe_units[0];
 
-            const auto exe_idx = (unsigned int) std::stoul(unit_count);
-            sparta_assert(exe_idx > 0, "Expected more than 0 units! " << tgt_name);
-            for(uint32_t unit_num = 0; unit_num < exe_idx; ++unit_num)
-            {
-                const std::string unit_name = tgt_name + std::to_string(unit_num);
-
-                // Create an InPort and an OutPort for credits and
-                // instruction send
-                auto & in_credit_port = in_credit_ports_.emplace_back
-                    (new sparta::DataInPort<uint32_t>(&unit_port_set_, "in_"+unit_name+"_credits"));
-                in_credit_port->enableCollection(node);
-
-                auto & out_inst_port = out_inst_ports_.emplace_back
-                    (new sparta::DataOutPort<InstQueue::value_type>(&unit_port_set_, "out_"+unit_name+"_write"));
-
-                // Create a Dispatcher for this target type
-                const auto target_itr = InstArchInfo::dispatch_target_map.find(tgt_name);
-                sparta_assert(target_itr != InstArchInfo::dispatch_target_map.end(),
-                              "Unknown target unit: " << tgt_name << " when parsing the execution_topology extension");
-                dispatchers_[target_itr->second].emplace_back(new Dispatcher(unit_name,
+            for(auto opt: core_types::issue_queue_types){
+                size_t found = exe_unit_first.find(opt);
```

Comment:
`const`

Reply:
Resolved in [6b0e363](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/6b0e3633c1b7331170afda8920f596f9fd16b233)